subsys/logging/log_core.c

```diff
@@ -282,7 +282,11 @@ static uint32_t z_log_init(bool blocking, bool can_sleep)
 				log_backend_enable(backend,
 						   backend->cb->ctx,
 						   CONFIG_LOG_MAX_LEVEL);
-			} else {
+			}
+			/* if the backend is activated by the if code above,
+			 * set the mask
+			 */
+			if (log_backend_is_ready(backend)) {
```

Comment:
why this change is needed?

Reply:
Previously, log_backend_is_ready is checked only once, if the backend is not ready, it will try enable it. If it is ready, it will set the mask. When the backend is enabled, it will miss the setting of the mask, so I redo the check after that. 
If you think the below code looks better, I can replace with it 
···
                        if (log_backend_is_ready(backend) == 0) {
                                log_backend_enable(backend,
                                                   backend->cb->ctx,
                                                   CONFIG_LOG_MAX_LEVEL);
                                if (log_backend_is_ready(backend)) {
                                        mask |= BIT(i);
                                }
                        }
                        else {
                                         mask |= BIT(i);
                        }
···